File: neptun_alarm_app/lib/config/prefs_keys.dart
/// Централізовані ключі для SharedPreferences
class PrefsKeys {
  PrefsKeys._();

  // ===== Налаштування сповіщень =====
  static const String notificationsEnabled = 'notifications_enabled';
  static const String subscribedRegions = 'subscribed_regions';
  static const String subscribedDistricts = 'subscribed_districts';
  static const String fcmToken = 'fcm_token';
  static const String deviceId = 'device_id';

  // ===== Тема та UI =====
  static const String darkMode = 'dark_theme';
  static const String firstLaunch = 'first_launch';
  static const String welcomeSkipped = 'welcome_skipped';

  // ===== TTS (Text-to-Speech) =====
  static const String ttsEnabled = 'tts_enabled';
  static const String ttsVolume = 'tts_volume';
  static const String ttsLanguage = 'tts_language';

  // ===== Вібрація =====
  static const String vibrationEnabled = 'vibration_enabled';
  static const String vibrationPattern = 'vibration_pattern';

  // ===== Sleep Mode =====
  static const String sleepModeEnabled = 'sleep_mode_enabled';
  static const String sleepModeStart = 'sleep_mode_start';
  static const String sleepModeEnd = 'sleep_mode_end';

  // ===== Premium =====
  static const String isPremium = 'is_premium';
  static const String premiumExpiry = 'premium_expiry';

  // ===== Ads =====
  static const String adFreeUntil = 'ad_free_until';

  // ===== Чат =====
  static const String chatUserId = 'chat_user_id';
  static const String chatNickname = 'chat_nickname';
  static const String chatAgeConfirmed = 'chat_age_confirmed';
  static const String chatRulesAgreed = 'chat_rules_agreed';
  static const String chatLikedMessages = 'chat_liked_messages';
  static const String isChatModerator = 'is_chat_moderator';
  static const String chatOfflineQueue = 'chat_offline_queue';

  // ===== Changelog =====
  static const String changelogLastShownVersion =
      'changelog_last_shown_version';

  // ===== Review =====
  static const String reviewAppLaunchCount = 'app_launch_count';
  static const String reviewFirstLaunchDate = 'first_launch_date';
  static const String reviewHasShownPrompt = 'has_shown_review_prompt';

  // ===== Медична картка =====
  static const String medicalBloodType = 'medical_blood_type';
  static const String medicalAllergies = 'medical_allergies';
  static const String medicalMedications = 'medical_medications';
  static const String medicalEmergencyContact1 = 'medical_emergency_contact_1';
  static const String medicalEmergencyContact2 = 'medical_emergency_contact_2';

  // ===== Тривожна валіза =====
  static const String emergencyBagItems = 'emergency_bag_items';

  // ===== Battery optimization (Xiaomi/Huawei) =====
  static const String batteryOptPromptLastShown = 'battery_opt_prompt_last_shown';

  // ===== Alarm tracking =====
  static const String lastAlarmState = 'last_alarm_state';
  static const String alarmStartTime = 'alarm_start_time';

  // ===== PRO: Custom alarm sound =====
  static const String alarmSoundId = 'pref_alarm_sound_id';
}

File: neptun_expo_app/src/config/prefsKeys.ts
/**
 * Mirrors Flutter `lib/config/prefs_keys.dart` for storage parity during migration.
 */
export const PrefsKeys = {
  selectedRegions: 'selected_regions',
  notificationsEnabled: 'notifications_enabled',
  subscribedRegions: 'subscribed_regions',
  subscribedTopics: 'subscribed_topics',
  subscribedDistricts: 'subscribed_districts',
  fcmToken: 'fcm_token',
  deviceId: 'device_id',

  darkMode: 'dark_theme',
  themeMode: 'app_theme_mode',
  firstLaunch: 'first_launch',
  welcomeSkipped: 'welcome_skipped',

  ttsEnabled: 'tts_enabled',
  ttsVolume: 'tts_volume',
  ttsLanguage: 'tts_language',

  vibrationEnabled: 'vibration_enabled',
  vibrationPattern: 'vibration_pattern',

  sleepModeEnabled: 'sleep_mode_enabled',
  sleepModeStart: 'sleep_mode_start',
  sleepModeEnd: 'sleep_mode_end',

  isPremium: 'is_premium',
  premiumExpiry: 'premium_expiry',
  activeTier: 'active_tier',
  debugPremium: 'debug_premium',

  adFreeUntil: 'ad_free_until',
  historyUnlockUntil: 'history_unlock_until',

  chatUserId: 'chat_user_id',
  chatNickname: 'chat_nickname',
  chatAgeConfirmed: 'chat_age_confirmed',
  chatRulesAgreed: 'chat_rules_agreed',
  chatLikedMessages: 'chat_liked_messages',
  isChatModerator: 'is_chat_moderator',
  chatOfflineQueue: 'chat_offline_queue',

  changelogLastShownVersion: 'changelog_last_shown_version',

  reviewAppLaunchCount: 'app_launch_count',
  reviewFirstLaunchDate: 'first_launch_date',
  reviewHasShownPrompt: 'has_shown_review_prompt',

  medicalBloodType: 'medical_blood_type',
  medicalAllergies: 'medical_allergies',
  medicalMedications: 'medical_medications',
  /** Flutter `safety_page.dart` keys */
  emergencyContact1: 'emergency_contact_1',
  emergencyContact2: 'emergency_contact_2',

  emergencyBagItems: 'emergency_bag_items',

  telegramRadarBannerDismissed: 'telegram_radar_banner_dismissed',
  radarSelectedFilter: 'radar_selected_filter',
  radarRecentSearches: 'radar_recent_searches',
  radarFollowedIds: 'radar_followed_ids',
  radarMutedCategories: 'radar_muted_categories',

  batteryOptPromptLastShown: 'battery_opt_prompt_last_shown',

  lastAlarmState: 'last_alarm_state',
  alarmStartTime: 'alarm_start_time',

  alarmSoundId: 'pref_alarm_sound_id',

  /** Flutter `BriefingService._briefingDateKey` */
  briefingLastSentDate: 'briefing_last_sent_date',
} as const;

export type PrefsKey = (typeof PrefsKeys)[keyof typeof PrefsKeys];
